### PR TITLE
Add common CQRS infrastructure

### LINKED ISSUE #1: Add common CQRS infrastructure
### Description
Add common infrastructure for CQRS and async communication

### Requirements

- project must be runnable
- mediatr must be registered to the middleware with loading types using reflection

### Deliverables and requirements

- `IQuery` - interface for all query requests
- `ICommand` - interface for all command requests
- `IEvent` - interface for all events
- `EventBroker` - broker for rabbit mq
- `RabbitMqConnectionProvider` - connection provider of rabbit mq
- `EventSubscriber` - event subscriber

### Reference

Models : 
- https://github.com/sulton-max/Backend.ILA.Api-Comms.Learning/tree/main/src/N2.Microservices.Common.Core/Models

Registering mediator  : 
- https://github.com/sulton-max/Backend.ILA.Api-Comms.Learning/blob/main/src/N2.Microservices.Services.BankingService.Api/Configurations/HostConfigurations.Extensions.cs

### PR DESCRIPTION
**Pull request by botirjonabdumannonovv**
Closes #1

### CODE REVIEW

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Queries/IQuery.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+/// Represents a query in a CQRS (Command Query Responsibility Segregation) architecture.
+/// </summary>
+/// <typeparam name="TResult">The type of result returned by the query.</typeparam>
+public interface IQuery<out TResult> : IRequest<TResult>, IQuery
+{
+}
+
+/// <summary>
+/// Marker interface for queries in a CQRS architecture.
+/// </summary>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Queries/IQuery.cs:15]: Change to "Defines"

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastucture/Common/EventBus/Services/EventBusBackgroundService.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        return Task.WhenAll(eventSubscribers.Select(eventSubscriber =>
+            eventSubscriber.StartAsync(stoppingToken).AsTask()));
+    }
+
+    public override Task StopAsync(CancellationToken cancellationToken)
+    {
+        return Task.WhenAll(eventSubscribers.Select(eventSubscriber =>
+            eventSubscriber.StopAsync(cancellationToken).AsTask()));
+    }
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastucture/Common/EventBus/Services/EventBusBackgroundService.cs:19]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastucture/Common/EventBus/Services/EventSubscriber.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    public ValueTask StopAsync(CancellationToken token)
+    {
+        Channel.Dispose();
+        return ValueTask.CompletedTask;
+    }
+
+    protected abstract ValueTask<(bool Result, bool Redeliver)> ProcessAsync(TEvent @event,
+        CancellationToken cancellationToken);
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastucture/Common/EventBus/Services/EventSubscriber.cs:23]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastucture/Common/Settings/EventBusSubscriberSettings.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+/// <summary>
+/// Represents settings for an event bus subscriber.
+/// </summary>
+public abstract class EventBusSubscriberSettings
+{
+    /// <summary>
+    /// Gets or sets the number of messages that the subscriber should prefetch at once.
+    /// </summary>
+    public ushort PrefetchCount { get; set; }
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastucture/Common/Settings/EventBusSubscriberSettings.cs:11]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Events/IEventHandler.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+/// Represents a handler for a specific type of event.
+/// </summary>
+/// <typeparam name="TEvent">The type of event being handled.</typeparam>
+public interface IEventHandler<in TEvent> : IEventHandler, INotificationHandler<TEvent> where TEvent : IEvent
+{
+    
+}
+
+/// <summary>
+/// Marker interface for event handlers.
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Events/IEventHandler.cs:15]: Change to "Defines"

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Events/Event.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+/// <summary>
+/// Represents an implementation of the IEvent interface, defining the properties for an event.
+/// </summary>
+public class Event: IEvent
+{
+    public Guid Id { get; set; } = Guid.NewGuid();
+    
+    public DateTimeOffset CreatedTime { get; set; } = DateTimeOffset.UtcNow;
+    
+    public bool Redelivered { get; set; }
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Events/Event.cs:12]: Add comments

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Commands/ICommand.cs [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+/// Represents a command in a CQRS (Command Query Responsibility Segregation) architecture.
+/// </summary>
+/// <typeparam name="TResult">The type of result produced by executing the command.</typeparam>
+public interface ICommand<out TResult> : ICommand, IRequest<TResult>
+{
+    
+}
+
+/// <summary>
+/// Marker interface for commands.
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Common/Commands/ICommand.cs:15]: Change to "Defines"

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/EventBus/Services/IAsyncEventBusSubscriber.cs [hunk 8]
```diff
@@ -0,0 +1,9 @@
+namespace Feedback.Analyzer.Application.Common.EventBus.Services;
+
+/// <summary>
+/// Represents an interface for an asynchronous event bus subscriber.
+/// </summary>
+public interface IAsyncEventBusSubscriber
+{
+    
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/EventBus/Services/IAsyncEventBusSubscriber.cs:9]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/EventBus/Brokers/IEventSubscriber.cs [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+    /// <param name="token">The cancellation token.</param>
+    /// <returns>A <see cref="ValueTask"/> representing the asynchronous operation.</returns>
+    ValueTask StartAsync(CancellationToken token);
+
+    /// <summary>
+    /// Stops the event subscription asynchronously.
+    /// </summary>
+    /// <param name="token">The cancellation token.</param>
+    /// <returns>A <see cref="ValueTask"/> representing the asynchronous operation.</returns>
+    ValueTask StopAsync(CancellationToken token);
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/EventBus/Brokers/IEventSubscriber.cs:20]: Remove

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/appsettings.Development.json [hunk 10]
```diff
@@ -3,6 +3,9 @@
     "LogLevel": {
       "Default": "Information",
       "Microsoft.AspNetCore": "Warning"
+    },
+    "NotificationSubscriberSettings": {
+      "PrefetchCount": 1
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/appsettings.Development.json:8]: Remove

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by bunyodkamalov — APPROVED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastucture/Common/EventBus/Brokers/EventBusBroker.cs [hunk 11]
```diff
@@ -0,0 +1,18 @@
+using Feedback.Analyzer.Application.Common.EventBus.Brokers;
+using Feedback.Analyzer.Domain.Common.Events;
+using MediatR;
+
+namespace Feedback.Analyzer.Infrastucture.Common.EventBus.Brokers;
+
+public class EventBusBroker(IPublisher mediator) : IEventBusBroker
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastucture/Common/EventBus/Brokers/EventBusBroker.cs:7]: Add comment

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastucture/Common/EventBus/Brokers/EventBusBroker.cs [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+{
+    public ValueTask PublishLocalAsync<TEvent>(TEvent @event) where TEvent : IEvent
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastucture/Common/EventBus/Brokers/EventBusBroker.cs:9]: Change to `Event`

**Review by sulton-max — APPROVED** [review]
Very well done

**Review by rashidovasadbek — APPROVED** [review]

**Pull request closed without merging**